routes/authors.js

```diff
@@ -1,8 +1,89 @@
 const express = require("express");
 const generateId = require("uuid/v4");
+const authorsDB = require("../data/authors");
+const booksDB = require("../data/books");
 
 const router = express.Router();
 
 // Define your endpoints here
+router.get("/", (req, res, next) => {
+  let attribute = req.query.sort;
+
+  function compare(a, b) {
+    if (a[attribute] < b[attribute]) return -1;
+    if (a[attribute] > b[attribute]) return 1;
+    return 0;
+  }
+
+  if (attribute) {
+    res.json(authorsDB.slice(0).sort(compare));
+  } else {
+    res.json(authorsDB);
+  }
+});
+
+router.post("/", (req, res, next) => {
+  const id = req.body.id;
+  const name = req.body.name;
+  const website = req.body.website;
+
+  const newAuthor = {
+    id: id,
+    name: name,
+    website: website
+  };
+
+  authorsDB.push(newAuthor);
+  res.json(newAuthor);
+});
+
+router.get("/:id", (req, res, next) => {
+  const author = authorsDB.filter(el => {
```

Comment:
`filter` will return an array. What you want here is an object, so you can use `find` instead.

By the way, you will need to handle the case where you don't find the author.